### PR TITLE
API test for downloadFromShare

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds more tests for download from share

## Related Issue
#31531 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 1]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
 		And the downloaded content when downloading file "/textfile0 (2).txt" for user "user2" with range "bytes=0-8" should be "BLABLABLA"
 		And user "user2" should see the following elements
 			| /common/sub/textfile0.txt |
-			| /textfile0%20(2).txt      |
\ No newline at end of file
+			| /textfile0%20(2).txt      |
+
+	Scenario: Downloading file inside the folder which is inside the user shared folder
```
Comment (patrickjahns) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: This is not quite clear to me - do you have an example?

Comment (phil-davis) [reply tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: Download a file that is in a folder contained in the shared folder

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 2]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
...
+		Given user "user0" has been created
+		And user "user1" has been created
+		And user "user0" creates a share using the API with settings
+			| path | PARENT     |
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: Make the middle | line up also

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 3]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
...
+			| shareType | 0     |
+			| shareWith | user1 |
+		Then the user "user1" should be able to download the file "/CHILD/child.txt" using the API
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: As discussed - this should give the full path like "/PARENT (2)/CHILD/child.txt"

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 4]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
...
+
+	Scenario: Downloading file inside the folder which is inside the group shared folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: Download a file that is in a folder contained in a folder that has been shared with a group

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 5]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
...
+		Given user "user0" has been created
+		And user "user1" has been created
+		And group "sharegroup" has been created
+		And user "user1" has been added to group "sharegroup"
+		And user "user0" has shared folder "PARENT" with group "sharegroup"
+		Then the user "user1" should be able to download the file "/CHILD/child.txt" using the API
+
+	Scenario: Downloading file inside the folder which is inside the public shared folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: Download a file that is in a folder contained in a folder that has been shared with public

File: tests/acceptance/features/apiSharing-v1/downloadFromShare.feature [hunk 6]
```diff
@@ -33,4 +33,60 @@ Feature: sharing
...
+		Given user "user0" has been created
+		When user "user0" creates a share using the API with settings
+			| path         | PARENT   |
+			| shareType    | 3        |
+			| password     | publicpw |
+		Then the public should be able to download the range "bytes=1-7" of file "/CHILD/child.txt" from inside the last public shared folder with password "publicpw" and the content should be "wnCloud"
+
+	Scenario: Downloading file inside the folder which is inside the user shared folder with share permission read/write
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/downloadFromShare.feature]: Make similar changes here to the words

**Comment by phil-davis** [comment]
I think the first scenarios already default to read/write. So we should mention that.
And the last scenarios need to explicitly do read-only.

**Review by phil-davis — APPROVED** [review]
Note: ``createShare.feature`` verifies that the share was created by testing that a file can be downloaded from the root folder of the share. So that means that ``createShare`` effectively has scenarios that test creating a share and being able to download from the share as a single scenario.

For that reason, the scenarios here in ``downloadFromShare`` are testing the case of downloading from a sub-folder inside the share - which is something different that needs to be tested.

**Comment by paurakhsharma** [comment]
Backport on #31459

**Pull request merged**